src/lib/storage/chunk.cpp

```diff
@@ -15,26 +15,30 @@
 namespace opossum {
 
 void Chunk::add_segment(const std::shared_ptr<AbstractSegment> segment) {
-  // Implementation goes here
+  DebugAssert(std::all_of(_segments.cbegin(), _segments.cend(),
+                          [&segment](std::shared_ptr<AbstractSegment> existing_segment) {
+                            return existing_segment->size() == segment->size();
+                          }),
+              "All segments should have the same size");
+  _segments.push_back(segment);
 }
 
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
+  DebugAssert(values.size() == column_count(), "The values to append have the same count as columns");
```

Comment:
If `values.size() < column_count()` in a release build, only a subset of the segments would be filled with values.
Using an `Assert` would prevent this.